File: 283.moveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int size = nums.size();
        int slow = 0;
        int fast = 1;

        while (fast < size) {
            while(fast < size && nums[fast] == 0) {fast++;}
            if (fast >= size) break;
            while(slow < fast && nums[slow] != 0) {slow++;}
            if (slow >= fast) {slow++;fast++;continue;}

            int tmp = nums[slow];
            nums[slow] = nums[fast];
            nums[fast] = tmp;
        }

    }
};

File: 283.moveZeroes.go
func moveZeroes(nums []int)  {
    length := len(nums)
    temp := 0
    if length == 0 {
        return
    }
    for i := 0; i < length; i++ {
        if nums[i] != 0 {
            nums[i], nums[temp] = nums[temp], nums[i]
            temp++
        }
    }
    return
}